src/webpack/CompilerEventsFacade.ts

```diff
@@ -5,36 +5,36 @@ export default class CompilerEventsFacade {
 
   private _compiler: Compiler;
 
-  private _legacyTapable: boolean;
-
   constructor(compiler: Compiler) {
     this._compiler = compiler;
-    this._legacyTapable = !compiler.hooks;
   }
 
   public afterOptimizeChunkAssets(
     call: (compilation: Compilation, chunks: Compilation["chunks"]) => void,
   ) {
-    return this._legacyTapable
-      ? (this._compiler as any).plugin("compilation",comp =>
-          comp.plugin("after-optimize-chunk-assets",chunks =>
-            call(comp, chunks),
-          ),
-        )
-      : this._compiler.hooks.compilation.tap(
-          CompilerEventsFacade.extensionName,comp =>
-            comp.hooks.afterOptimizeChunkAssets.tap(
-              CompilerEventsFacade.extensionName,chunks => call(comp, chunks),
-            ),
+    return this._compiler.hooks.compilation.tap(
+      CompilerEventsFacade.extensionName,
+      (comp) => {
+        const afterOptimizeChunkAssets = (chunks) => {
```

Comment:
Can we rename this function so there is no namespace overlap? 

Additionally, I am not sure if this passes in the same chunks via the tap, is there a reason we are taking this approach?